Review of Update flow in Mbed OS reference docs

melwee01 opened the pull request.

Commit:
adding more notes

AnotherButler commented on docs/reference/technology/firmware_update.md:
```diff
@@ -21,31 +22,34 @@ The active firmware, made up of Mbed OS and an application, can be updated only
 - The application starts immediately after the bootloader.
 
 If the Mbed tool finds a manageable bootloader, the image build process automatically merges the bootloader image with the application image.
+>We say managed and manageable--which one should we go with?
```
"Managed." We have "managed" and "unmanaged." "Manageable" and "unmanageable" would have very different connotations :)

AnotherButler commented on docs/reference/technology/firmware_update.md:
```diff
@@ -54,19 +58,22 @@ For more information about [managed and unmanaged bootloaders, and how to create
 
 If you're interested in porting the Update client and bootloader to new hardware, please [review the porting section](https://www.pelion.com/docs/device-management/current/porting/porting-the-device-management-update-client.html).
-
+>Can this link just point to the DM-side porting docs?
```
If this is all about porting a bootloader with Update client and we have a guide for that, I think we should link to that guide as opposed to generic porting.

melwee01 replied:
The Update client portion of the porting guide might go--if you port Client, Update client is included with it. So there's no point in having separate docs for 'porting' it. There might be additional material on *configuration*, if necessary. I'll change this link as necessary.

AnotherButler reviewed (commented):
I left some initial responses to some of the queries.

yogpan01 commented:
IMHO, this page should not even exist here as Update feature is a part of Device Management Client and not necessarily part of MbedOS docs.
Update works on every PDMC supported platforms including MbedOS, Linux and Unisoc.
It is not unique for MbedOS and there is no code inside MbedOS which does Update so why are we giving information to developers that Update is somehow intrinsic part of MbedOS. It clearly is not.

yogpan01 commented:
@iriark01 Please consider removing this page from mbed-os side we are covering the Update usage extensively on device management and right thing to do would be to redirect users to that site instead of duplicating information here.

40Grit commented:
It would be nice to have a page regarding non-pelion initiated updates.

That might be more mbed-bootloader territory though...

Commit:
more editorial changes to firmware update in OS

Commit:
formatting and spelling, taking out redundant bootloader info

Commit:
removing one comment

c1728p9 commented on docs/reference/technology/firmware_update.md:
```diff
@@ -1,96 +1,81 @@
+Mbed OS integrates Pelion Device Management firmware update services, so if you have a Device Management account, you can send binaries to your remote device.
 
-Update support in Mbed OS relies on two features:
+Mbed OS now includes the boot loader that manages update verification and installation.
```
Might want to be more specific if you are talking about the Pelion bootloader. Maybe something like
```
that manages update verification and installation for Pelion Client.
```

c1728p9 commented on docs/reference/technology/firmware_update.md:
```diff
@@ -1,96 +1,81 @@
+## The Mbed OS boot loader
 
-During system startup, a bootloader is responsible for selecting and forwarding control to the next stage in the boot sequence. The Mbed OS bootloader also validates the next stage's sanity and signature before forwarding control to it; if validation fails, the bootloader can select an alternate version of the next stage. It can also install an alternative upon request (in other words, firmware u ...
+During system startup, a boot loader is responsible for selecting and forwarding control to the next stage in the boot sequence. The Mbed OS boot loader also validates the next stage's sanity and signature before forwarding control to it; if validation fails, the boot loader can select an alternate version of the next stage. It can also install an alternative upon request (in other words, firmwar ...
```
May want to clarify that this is the Pelion bootloader.

c1728p9 commented on docs/reference/technology/firmware_update.md:
```diff
@@ -1,96 +1,81 @@
+A boot sequence can have several stages of boot loaders, leading to an application. The different stages (including the application) may need to evolve over time, to add features or bug-fixes. Upgrades are possible for boot sequences with two or more stages: any active stage can replace the next stage in the sequence; when the system restarts, it executes the updated components. Typically, howeve ...
 
-To protect against faults in the newly updated components, we store multiple versions (up to a certain maximum number of versions). Each stag is responsible for detecting faults in the following component and rolling it back if it discovers faults. For example, if stage 3 is unstable, needs additional functionality or is otherwise behaving incorrectly, stage 2 (during the *next* startup sequence) ...
+To protect against faults in the newly updated components, we store multiple versions (up to a certain maximum number of versions). Each stage is responsible for detecting faults in the following component and rolling it back if it discovers faults. For example, if the third stage is unstable, needs additional functionality or is otherwise behaving incorrectly, the second stage (during the *next* ...
```
This makes it sound like Mbed/Pelion has a 3+ stage bootloader. As far as I know both only use 2 stages (bootloader + application).

cc @yogpan01

c1728p9 commented on docs/reference/technology/firmware_update.md:
```diff
@@ -1,96 +1,81 @@
 1. Permission to access your Device Management account, as well as keys and certificates to verify the firmware's source and validity.
 
-## The Mbed OS bootloader
+## The Mbed OS boot loader
```
The phrase `The Mbed OS boot loader` is a bit vague. I would be specific on the separate concepts:

- Mbed OS bootloader framework
- Pelion Bootloader - [mbed-bootloader](https://github.com/ARMmbed/mbed-bootloader)
- Mbed OS Bootloader Example - [mbed-os-example-bootloader](https://github.com/ARMmbed/mbed-os-example-bootloader)

Commit:
Clarify content in firmware_update.md

Clarify content to show this page is talking about the Pelion bootloader, not the example bootloader or the Mbed OS bootloader framework.

iriark01 committed:
Fixing callback URL

Commit:
Fix link in iBeacon.md

Fix broken link for building.

Commit:
Fix link in Watchdog.md

Fix broken link for publishing.

Commit:
Fix link in Watchdog.md

Fix broken link.

Commit:
Pelion -> Device Management

Commit:
taking out bootloader

Commit:
links

Commit:
Edit firmware_update.md

Edit file, mostly for active voice and precise language.

AnotherButler commented:
Thanks for the update. Please rebase to resolve merge conflicts.

AnotherButler commented:
@melwee01 Could you please rebase?

AnotherButler commented:
Closing because it was easier to make changes manually than rebase this time.
